avro-codegen/src/main/java/com/linkedin/avroutil1/codegen/SpecificRecordClassGenerator.java

```diff
@@ -33,8 +70,35 @@
 public class SpecificRecordClassGenerator {
 
   private static final String AVRO_GEN_COMMENT = "GENERATED CODE by avro-util";
+  private int sizeValCounter = 0;
 
-  public JavaFileObject generateSpecificRecordClass(AvroNamedSchema topLevelSchema, SpecificRecordGenerationConfig config) {
+
+  HashSet<TypeName> fullyQualifiedClassNamesInRecord = new HashSet<>();
+  HashSet<Class<? extends Object>> fullyQualifiedClassesInRecord = new HashSet<>(Arrays.asList(
```

Comment:
could "pre-loading" this with classes cause unused imports in the generated class if we later dont use any of these? (for example due to settings)

Reply:
These classNames are added to the always Qualify list. It will force full qualification of these classes when used. Haven't seen it leading to any unused imports yet.